### PR TITLE
Support upstream jdk sha

### LINKED ISSUE #3309: Support commitSHA as Tag
For reproducible build comparison we'd like to reproduce the nightly build jdk by checking out the specific commit. Currently the build only support a custom branch using the commit will get  the fatal error `fatal: Couldn't find remote ref refs/tags/f5445fd6891bff118b2b92d00d85c7b22b7927b7`

### PR DESCRIPTION
**Pull request by sophia-guo**
Support openjdk commitID as TAG for both release and non-release build

Close #3309 

Signed-off-by: Sophia Guo <sophia.gwf@gmail.com>

### CODE REVIEW

File: sbin/prepareWorkspace.sh [hunk 1]
```diff
@@ -165,7 +167,20 @@ checkoutRequiredCodeToBuild() {
...
+          git reset --hard || rc=$?
+          if [ $rc -eq 0 ]; then
+            echo "Checked out tag ${tag}"
+          else
+            echo "Failed cmd: git reset --hard"
+          fi
+        else
+          echo "Failed cmd: git checkout \"${tag}\""
+        fi
+        echo "Error Failed cmd: git checkout \"${tag}\""
```
Comment (andrew-m-leonard) [inline sbin/prepareWorkspace.sh:183]: I think echo this line shouldn't be there

File: sbin/prepareWorkspace.sh [hunk 2]
```diff
@@ -165,7 +167,20 @@ checkoutRequiredCodeToBuild() {
           echo "Failed cmd: git checkout \"${tag}\""
         fi
       else
-        echo "Failed cmd: git fetch origin \"refs/tags/${tag}:refs/tags/${tag}\""
+        echo "Checking out commit id ${tag}"
```
Comment (andrew-m-leonard) [inline sbin/prepareWorkspace.sh:170]: Perhaps change this echo to "Unable to checkout as refs/tags/${tag}, attempting to checkout as commit id $tag"

File: sbin/prepareWorkspace.sh [hunk 3]
```diff
@@ -179,8 +194,6 @@ checkoutRequiredCodeToBuild() {
           else
             echo "Failed cmd: git reset --hard \"origin/${BUILD_CONFIG[BRANCH]}\""
           fi
-        else
-          echo "Failed cmd: git fetch --all ${BUILD_CONFIG[SHALLOW_CLONE_OPTION]}"
```
Comment (andrew-m-leonard) [inline sbin/prepareWorkspace.sh:183]: Should this have been deleted? fetch --all will fail silently

**Comment by sophia-guo** [comment]
Close in favour of https://github.com/adoptium/temurin-build/pull/3386

**Pull request closed without merging**